src/railfence.rs

```diff
@@ -0,0 +1,284 @@
+        //      .x.x.x.x.x.x.
+        //      ..x...x...x..
+        //   The ciphertext is then written onto the indentified positions, line by line
+        //      H...o...o...!
+        //      .e.l.,.W.r.d.
+        //      ..l... ...l..
+        //   The decrypted message is then read in a zigzag:
+        //      Hello, World!
+        
+
+        // A key of one causes a problem when calculating the cycle later on,
+        // so just return the decrypted message now
+        if self.key == 1 {
+            return cipher_text.to_string()
+        }
+
+        // Create the table that will be used for decryption
+        // The form of an entry is (bool, char).
+        // The bool determines whether the current entry is being used, and if so
+        // the char is part of the plain/cipher text
+        let mut table = vec![vec![(false, 'a'); cipher_text.len()]; self.key];
+
+        // Find elements in the table that should be filled
+        for i in 0..cipher_text.len() {
+            let col = i;
+            // In the railfence cipher the letters are placed diagonally in a zigzag,
+            // so, with a key of 4 say, the row numbers will go
+            //      0, 1, 2, 3, 2, 1, 0, 1, 2, 3, 2, 1, 0, ...
+            // This repeats with a cycle (or period) given by (2*key - 2)
+            //      [0, 1, 2, 3, 2, 1], [0, 1, 2, 3, 2, 1], 0, ...
+            // This cycle is always even.
+            let cycle = 2 * self.key - 2;
+            // For the first half of a cycle, the row is given by the index,
+            // but for the second half it decreases and is therefore given by the reverse index,
+            // the distance from the end of the cycle.
+            let row = if i % cycle <= cycle / 2 {
+                i % cycle
+            }
+            else {
+                cycle - i%cycle
+            };
+
+            // Fill cell with an arbitrary letter
+            table[row][col] = (true, 'a');
+        }
+
+        // Fill the identified positions in the table with the ciphertext, line by line
+        let mut ct_iter = cipher_text.chars();
+        for row in table.iter_mut() {
+            for entry in row.iter_mut() {
+                if entry.0 {
+                    *entry = (true, ct_iter.next().unwrap());
+                }
+            }
+        }
+
+        // Read the plaintext in a zigzag
+        let mut message = String::new();
+        for i in 0..cipher_text.len() {
+            let col = i;
+            let cycle = 2 * self.key - 2;
+            let row = if i % cycle <= cycle / 2 {
+                i % cycle
+            }
+            else {
+                cycle - i%cycle
+            };
+
+            message.push(table[row][col].1);
+        }
+
+        message
+    }
+
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn encrypt_test() {
+        let key = 6;
+        let message = "attackatdawn";
+        let r = Railfence::new(key).unwrap();
```

Comment:
To keep the style standard across the tests, could you please init the Railfence cipher directly rather than creating a separate key? e.g. `Railfence::new(6).unwrap`